File: LowLevelDesign/java/tictactoe/src/com/tictac/entities/Cell.java
package com.tictac.entities;

import com.tictac.enums.Symbol;

public class Cell {
    private final int row;
    private final int col;
    private Symbol symbol;
    public Cell(int row, int col) {
        this.row = row;
        this.col = col;
    }

    public void updateCell(Symbol symbol) {
        this.symbol = symbol;
    }

    public boolean isEmpty() {
        return this.symbol == null;
    }

    public Symbol getSymbol() {
        return this.symbol;
    }
}

File: SystemDesignDiagrams/Interviews/tic-tac-toe/entities/Cell.js
export class Cell {
    constructor(row, col) {
        this.row = row;
        this.col = col;
        this.symbol = null; // X, O
    }

    updateCell(symbol) {
        this.symbol = symbol;
    }

    getSymbol() {
        return this.symbol;
    }
}